Language/Structure/Control Structure/ternary.adoc

```diff
@@ -0,0 +1,71 @@
+---
+title: ternary
+categories: [ "Structure" ]
+subCategories: [ "Control Structure" ]
+---
+
+
+
+
+
+= ?: Ternary Operator
+
+
+// OVERVIEW SECTION STARTS
+[#overview]
+--
+
+[float]
+=== Description
+It takes three arguments rather than the typical one or two that most operators use. The first argument is a comparison argument, the second is the result upon a true comparison, and the third is the result upon a false comparison. It is an alternative to shorten an if-else block. It can help increase the readability and reduce the number of lines in your code.
+[%hardbreaks]
+
+
+[float]
+=== Syntax
+`expression_1 ? expression_2 : expression_3; // if expression_1 evaluates to true then expression_2 is evaluated else expression_3`
+
+
+[float]
+=== Parameters
+`expression`. An expression is any legal combination of symbols that represents a value.
+
+--
+// OVERVIEW SECTION ENDS
+
+
+
+// HOW TO USE SECTION STARTS
+[#howtouse]
+--
+
+[float]
+=== Example Code
+
+[source,arduino]
+----
+int x = 10, y = 20, z; // declared and defined three variables x, y and z
+z = (x < y) ? x : y;
```

Comment:
```suggestion
int x = 10;
int y = 20;
int z = (x < y) ? x : y;
```

Beginners are likely to be unfamiliar with the ability to declare multiple variables in one statement. This is not documented in the Arduino Language Reference and they won't understand that the language contains all the capabilities of C++ and that the missing documentation for those capabilities can be found in any C++ reference, since this fact is not documented (https://github.com/arduino/reference-en/issues/623).

In this case, using a separate statement for each declaration is actually just as good because the function of the code is now so clear that an explanatory comment is unnecessary.